packages/core/src/components/ApplicationLauncher/ApplicationLauncherToggle.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { OverlayTrigger } from '../OverlayTrigger';
+import { Tooltip } from '../Tooltip';
+import { Icon } from '../Icon';
+import { Button } from '../Button';
+
+const ApplicationLauncherToggle = ({
+  open,
+  tooltip,
+  onClick,
+  tooltipPlacement
+}) => {
+  if (tooltip)
+    return (
+      <OverlayTrigger
+        placement={tooltipPlacement}
+        id="applauncher-pf-block-list"
+        overlay={<Tooltip>{tooltip}</Tooltip>}
+      >
+        <Button
+          onClick={onClick}
+          bsStyle="link"
+          className="nav-item-iconic dropdown-toggle"
+          aria-describedby="tooltip"
```

Comment:
Where is this value coming from? This should the the same value that displays as the tooltip `id` value, but it isn't obvious to me where the tooltip `id` value is being defined. When I look at the [react-bootstrap documentation](https://react-bootstrap.github.io/components/tooltips/), they show examples of tooltip like this `<Tooltip id={id}>` with `id` as a required prop.

Reply:
yeah, ill just change it:
`        overlay={<Tooltip id={tooltip}>{tooltip}</Tooltip>}
`
in the `OverlayTrigger`